hr_holidays_accrual_advanced/wizards/hr_leave_allocation_accrual_calculator.py

```diff
@@ -45,32 +42,27 @@ class HrLeaveAllocationAccrualBalanceCalculator(models.TransientModel):
             'hr.leave.allocation.accrual.calculator.accruement'
         ),
         inverse_name='calculator_id',
-        readonly=True,
     )
     accrued = fields.Float(
         string='Accrued',
-        readonly=True,
     )
     balance = fields.Float(
         string='Balance',
-        readonly=True,
     )
 
     @api.onchange(
         'date',
     )
     def _onchange(self):
-        self._recalculate()
-
-    @api.multi
-    def _recalculate(self):
-        self.ensure_one()
+        CalculatorAccruement = self.env[
+            'hr.leave.allocation.accrual.calculator.accruement'
+        ]
+        accruement_ids = CalculatorAccruement
 
         if not self.date:
-            self.write({
-                'balance': 0.0,
-                'accruement_ids': [(5, False, False)],
-            })
+            self.accrued = 0.0
```

Comment:
Why not writing in block?

Reply:
You can not write many2many relation on non-saved record without crash